src/Pages/DrinkDetails.jsx

```diff
@@ -14,13 +16,89 @@ class DrinkDetails extends Component {
       ingredient: [],
       measure: [],
       recomandation: [],
+      copyToClipboard: false,
+      favoriteFood: false,
     };
     this.fetchDetail = this.fetchDetail.bind(this);
     this.renderRecomendations = this.renderRecomendations.bind(this);
+    this.saveOnLocalStorage = this.saveOnLocalStorage.bind(this);
+    this.saveOnLocalStorage = this.saveOnLocalStorage.bind(this);
+    this.addMealInProgress = this.addMealInProgress.bind(this);
+    this.verifyRecipeIsDone = this.verifyRecipeIsDone.bind(this);
+    this.saveFavoriteRecipes = this.saveFavoriteRecipes.bind(this);
   }
 
   componentDidMount() {
     this.fetchDetail();
+    const { match: { params: { id } } } = this.props;
+    this.newFunction = () => {
+      const favoriteRecipes = JSON.parse(localStorage.getItem('favoriteRecipes'));
+      let favRecipeLocalStorage;
+      if (favoriteRecipes !== null) {
+        const favRecipenew = favoriteRecipes[0].id;
+        favRecipeLocalStorage = favRecipenew;
+      }
+      if (favRecipeLocalStorage === id) {
+        this.setState({
+          favoriteFood: true,
+        });
+      }
+    };
+    this.newFunction();
+  }
+
+  verifyRecipeIsDone() {
+    const { match: { params: { id } } } = this.props;
+    const doneRecipes = JSON.parse(localStorage.getItem('doneRecipes')) || [];
+    if (doneRecipes.length > 0) {
+      return doneRecipes.some((recipe) => recipe.id === id);
+    }
+    return false;
+  }
+
+  addMealInProgress() {
+    const { match: { params: { id } } } = this.props;
+    const inProgress = { cocktails: {}, meals: {} };
+    localStorage.setItem('inProgressRecipes', JSON.stringify(inProgress));
```

Comment:
Mano, o que voce acha de alterar essa linha, pra caso a chave existir, ele não sobrescrever no LocalStorage?
```suggestion
     if (localStorage.getItem('inProgressRecipes') === null) {
        localStorage.setItem('inProgressRecipes', JSON.stringify(inProgress));
      }
```